src/main/java/com/dataguise/saas/service/ConnectionService.java

```diff
@@ -0,0 +1,450 @@
+        throws BadRequestException, BadGatewayException, InternalServerException {
+        DgConnection conn = connectionDTO.getConnection();
+        Boolean defaultSchedule = connectionDTO.getAutoSchedule();
+        String agentHostName = connectionDTO.getAgentHostName();
+        String agentName = connectionDTO.getAgentName();
+
+        String connectionAgentType = conn.getConnectionAgent();
+        Integer connectionId = null;
+        if (connectionAgentType == null) {
+            throw new BadRequestException("Invalid Connection Agent type");
+        }
+        if (connectionAgentType
+            .equalsIgnoreCase(Connection.ConnectionType.Discover.getConnectionType())) {
+            List<String> schemaFilters = conn.getSchemaFilter();
+            Iterator<String> itr = schemaFilters.iterator();
+            while (itr.hasNext()) {
+                String schema = itr.next();
+                if (schema.equals("")) {
+                    throw new BadRequestException(
+                        ErrorConstants.ERROR_VALIDATING_SCHEMA.getErrorMessage(),
+                        HttpStatus.BAD_REQUEST);
+                }
+            }
+            if (conn.getSchemaFilter().size() == 0) {
+                throw new BadRequestException(
+                    ErrorConstants.ERROR_VALIDATING_SCHEMA_FILTER.getErrorMessage(),
+                    HttpStatus.BAD_REQUEST);
+            }
+
+            if (conn.getGroupId() == null || conn.getGroupId() == 0) {
+                //Make sure UI sets group Id using getAgentDetails call.
+                throw new BadRequestException(
+                    ErrorConstants.ERROR_VALIDATING_GROUP_ID.getErrorMessage());
+            }
+
+            if (agentHostName == null || agentHostName.isEmpty()) {
+                //Make sure UI sets agentHostName using getAgentDetails call.
+                throw new BadRequestException(
+                    ErrorConstants.ERROR_VALIDATING_AGENT_HOSTNAME.getErrorMessage());
+            }
+
+            if (agentName == null || agentName.isEmpty()) {
+                //Make sure UI sets agentName using getAgentDetails call.
+                throw new BadRequestException(
+                    ErrorConstants.ERROR_VALIDATING_AGENT_NAME.getErrorMessage());
+            }
+        }
+        try {
+            if (conn.getConnectionType() != null) {
+                conn.setModule(restDriver.getModuleTypeUsingSourceType(conn.getConnectionType()));
+                connectionId = restDriver.createConnection(conn);
+            } else {
+                throw new BadRequestException(
+                    ErrorConstants.ERROR_CREATING_CONNECTION.getErrorMessage());
+            }
+        } catch (DgDrvException e) {
+            logger.error("Create connection failed: " + e);
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        logger.info("Connection created with id: " + connectionId);
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/ConnectionService.java L160
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1846854166>*